src/main/java/org/cascadebot/cascadebot/utils/votes/VoteButtonGroup.java

```diff
@@ -27,19 +37,27 @@
 
     private BiConsumer<List<VoteResult>, Message> periodicConsumer;
 
+    private Consumer<List<VoteResult>> finishConsumer;
+
     private Set<Long> allowedUsers;
 
     private Timer timer = new Timer();
 
     private Timer voteTimer;
 
-    VoteButtonGroup(long ownerId, long channelId, long guildId, BiConsumer<List<VoteResult>, Message> periodicRunnable, Timer voteTimer) {
+    private long timerStartTime;
+
+    private int timerRunTime = 10;
+
+    VoteButtonGroup(long ownerId, long channelId, long guildId, BiConsumer<List<VoteResult>, Message> periodicRunnable, Timer voteTimer, long timerStartTime, Consumer<List<VoteResult>> finishConsumer) {
```

Comment:
Why is the timer start an option? I could see it maybe as a separate method in case you have it in the future, but by default it should just be the system time, or the time the message is actuality sent (there's a consumer for that on button groups)

Reply:
I passed it through so it was the exact time it started but you're right this can be replaced by just system time